### PR TITLE
fix/beaxy trading pairs parsing

### LINKED ISSUE #4752: Beaxy fails to parse trading pair
**Describe the bug**
Similar to ticket #4750, when starting the a bot using Beaxy connector using PR #4678 some trading pairs encounter a parsing error.
```python
2021-11-18 10:53:32,426 - 3897 - hummingbot.connector.exchange.beaxy.beaxy_exchange - ERROR - Error parsing the trading_pair rule {'symbol': 'GOBTC', 'name': 'GOBTC', 'minimumQuantity': 50.0, 'maximumQuantity': 2000000.0, 'quantityIncrement': 0.001, 'quantityPrecision': 0, 'tickSize': 1e-08, 'baseCurrency': 'GO', 'termCurrency': 'BTC', 'pricePrecision': 8, 'buyerTakerCommissionProgressive': 0.25, 'buyerMakerCommissionProgressive': 0.15, 'sellerTakerCommissionProgressive': 0.25, 'sellerMakerCommissionProgressive': 0.15, 'type': 'crypto', 'suspendedForTrading': False}. Skipping.
Traceback (most recent call last):
  File "hummingbot/connector/exchange/beaxy/beaxy_exchange.pyx", line 814, in hummingbot.connector.exchange.beaxy.beaxy_exchange.BeaxyExchange._format_trading_rules
    retval.append(TradingRule(trading_pair,
  File "hummingbot/connector/trading_rule.pyx", line 37, in hummingbot.connector.trading_rule.TradingRule.__init__
    quote_token = split_hb_trading_pair(self.trading_pair)[1]
  File "/Users/clintg/Github/perp-mm/hummingbot/connector/utils.py", line 40, in split_hb_trading_pair
    base, quote = trading_pair.split("-")
ValueError: not enough values to unpack (expected 2, got 1)
```
**Steps To Reproduce**

Here are the steps to reproduce the issue (see attachments in the section below):
1. Create any strategy using Beaxy connector
2. Start the bot 
...

**Screenshots**
image

**Release version**
0.46.0

**Attachments**

[logs_conf_pure_mm_4.log](https://github.com/CoinAlpha/hummingbot/files/7560028/logs_conf_pure_mm_4.log)

*WARNING: Do NOT publish any exchange API keys or your wallet's private key. Whoever has access to them may steal your assets!*

**Optional: your discord username:**

### PR DESCRIPTION
**Pull request by aarmoa**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request:**
Change in Beaxy connector to keep a list of available tokens and use it to translate tokens from exchange format to client format back and forth. Since the connector is a spot connector the new logic excludes all perpetual tokens from the list.

**Tests performed by the developer:**
I have added unit tests for all the new functionality related to the population of the token list and the new functions to translate token formats.
I have not added tests for the connector functions that consume the translation methods because we need to send this fix to production with the next release. Also in the near future we will be working on replacing websockets library for aiohttp in Beaxy, and we will add all the required tests when tackling that task.

**Tips for QA testing:**
Please test the connector can create and cancel orders correctly, and the trade fills are processed without issues. I could not test the connector myself completely because the dev Beaxy account has not enough founds.

Fixes #4752

### CODE REVIEW

**Review by PtrckM — APPROVED** [review]
Tested and confirmed working, approving in behalf of QA.

File: hummingbot/connector/exchange/beaxy/beaxy_api_order_book_data_source.py
Comment (petioptrv) [inline hummingbot/connector/exchange/beaxy/beaxy_api_order_book_data_source.py:77]: Given that this map will store all pairs from the exchange, I propose we use a two-way map to get this to O(1). [This answer](https://stackoverflow.com/a/13276237/6793798) has a nice implementation, except that I don't see why the `del`'s are needed, so I think this should be sufficient:
```Python
class TwoWayDict(dict):
    def __setitem__(self, key, value):
        dict.__setitem__(self, key, value)
        dict.__setitem__(self, value, key)

    def __delitem__(self, key):
        dict.__delitem__(self, self[key])
        dict.__delitem__(self, key)

    def __len__(self):
        """Returns the number of connections"""
        return dict.__len__(self) // 2
```

File: hummingbot/connector/exchange/beaxy/beaxy_api_order_book_data_source.py
Comment (petioptrv) [inline hummingbot/connector/exchange/beaxy/beaxy_api_order_book_data_source.py:85]: I see that a two-way map won't work here. Maybe maintaining a forward and inverse map?

File: test/hummingbot/connector/exchange/beaxy/test_beaxy_api_order_book_data_source.py
Comment (petioptrv) [inline test/hummingbot/connector/exchange/beaxy/test_beaxy_api_order_book_data_source.py:45]: I propose we make one of the pairs something like `base = "A"; quote = "B"`, which will fail on the previous implementation, but will pass with this implementation and will demonstrate that any case can be handled now

**Review by petioptrv — APPROVED** [review]
Approving in case we don't have time for the suggested changes, but please consider them otherwise

**Comment by dennisocana** [comment]
@aarmoa I'm going to merge this PR after collecting the required approvals from QA and engineering so I can sync to staging. Should there be any more changes after this, please create a separate PR.

**Pull request merged**
